atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -516,6 +534,54 @@ private void unmarkStreamsAsUsed() {
                     line("index.delete(toDelete);");
                 } line("}");
             }
+
+            private void storeBlocksAndGetFinalMetadata() {
+                line("@Override");
+                line("protected StreamMetadata storeBlocksAndGetFinalMetadata(Transaction t, long id, InputStream stream) {"); {
+                    line("//Hash the data before compressing it");
+                    line("MessageDigest digest = Sha256Hash.getMessageDigest();");
+                    line("InputStream hashingStream = new DigestInputStream(stream, digest);");
+                    line("InputStream compressingStream;");
+                    line("try {"); {
```

Comment:
I think we want this to be try-with-resource for the compressingStream to ensure the streams are closed if an exception occurs during storeBlocks